### DISCUSSION TITLE
How to show defualt value in material UI multi select

### DISCUSSION THREAD

**Discussion by sivaputrevu** [react-hook-form/react-hook-form#2795] [Q&A] [answered]
Hi Bill,

Is there any way to set the default values for material UI multi-select?
in case of create the default should be empty and in case of edit I need to populate the previous selected fields.
defaultValue from MUI also not working

here is my code sandbox.
https://codesandbox.io/s/withered-bash-9vh86?fontsize=14&hidenavigation=1&theme=dark 

thanks,

**Comment by bluebill1049** [comment, marked as answer]
<img width="616" alt="Screen Shot 2020-09-02 at 4 07 38 pm" src="https://user-images.githubusercontent.com/10513364/91937968-6cf16e00-ed36-11ea-8003-69f5725d1d0d.png">
hey @sivaputrevu looks like you are not using render props API correctly, you shouldn't need to use setValue, please wire up all the props coming from render.

**Reply by sivaputrevu** [reply]
any example on this please?

**Reply by bluebill1049** [reply]
from the offcial doc: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by sivaputrevu** [reply]
I made the change instead of setValue(), it's working but how to display default values in the multi select material ui? defaultValue from select API not working. will the above change also works to set the defaults?

**Reply by bluebill1049** [reply]
find out your controlled component's default state, you can apply that to your controller's defaultValue. also, try to learn from the CSB above.

**Reply by sivaputrevu** [reply]
Hi Bill,
I figured it out! it is working as expected.
I set the default value for the multi select and the same populated into the render -> elementProps. using renderValue I am able to display the defaults and the same I have updated in the code sandbox.
thanks for replying as fast as you can! always you deserve an appreciation from me!!!! great job!!!

**Reply by bluebill1049** [reply]
That's awesome to hear! well done mate. "pad your shoulder"

**Comment by geomus** [comment]
Hi! @sivaputrevu could you please show me a snippet of the code that worked for you to see default values as Chips? thank you!

**Reply by bluebill1049** [reply]
Maybe this CSB is helpful: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

**Reply by sivaputrevu** [reply]
@geomus, please find the link below 
https://codesandbox.io/s/kind-glade-hughk?file=/src/App.tsx

**Reply by geomus** [reply]
Thank you both! @bluebill1049 @sivaputrevu  I was off a couple of days. I just made it through those links! I wrote with little hope I must say haha, thanks!

**Reply by MadhaneeswaranP** [reply]
Hai bill! @bluebill1049 I am trying to set default values for the checkbox, The default values are fetched from api. But I'm getting this error. Thank you in advance!
**Material-UI: The value provided to Autocomplete is invalid.**
**None of the options match with `""`.**
**You can use the `getOptionSelected` prop to customize the equality test. **


`import React from "react"`
`import Checkbox from "@material-ui/core/Checkbox"`
`import { Grid } from "@material-ui/core"`
`import { TextField as MuiTextField } from "@material-ui/core"`
`import Autocomplete from "@material-ui/lab/Autocomplete"`
`import CheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank"`
`import CheckBoxIcon from "@material-ui/icons/CheckBox"`

`const icon = <CheckBoxOutlineBlankIcon fontSize="small" />`
`const checkedIcon = <CheckBoxIcon fontSize="small" />`

`export default function StoreFilter({ store, data, onStoreChange }) {`
   `const [defaultStore, setDefaultStore] = React.useState(store)`

   `console.log("IN store filter default store value",store)`

  `return (`
  `<>`
      `<Grid container style={{ justifyContent: "flex-end" }}>`
       ` <Grid item style={{ marginTop: "15px", marginBottom: "20px" }}>`
         ` <Autocomplete`
            `multiple`
            `id="checkboxes-tags-demo"`
            `options={data}`
            `disableCloseOnSelect`
            `defaultValue={[defaultStore]}`
           ` getOptionLabel={(option) => (option ? option.name : "")}`
            `getOptionSelected={(option, value) => option.name === value.name}`
            `renderOption={(option, { selected }) => (`
             ` <React.Fragment>`
                `<Checkbox`
                 ` icon={icon}`
                 ` checkedIcon={checkedIcon}`
                `  style={{ marginRight: 8 }}`
                `  checked={selected}`
                `/>`
              `  {option.name}`
             ` </React.Fragment>`
           ` )}`
           ` style={{ width: 550 }}`
           `renderInput={(params) => (`
             ` <MuiTextField`
              `  {...params}`
               ` fullWidth`
               ` variant="outlined"`
               ` label="Stores"`
               ` placeholder="More..."`
             ` />`
          `  )}`
          `  onChange={(event, value) => {`
            `  onStoreChange(value)`
            `  setDefaultStore(null)`
          `  }}`
         ` />`
       ` </Grid>`
      `</Grid>`
   ` </>`
 ` )`
`}`

**Comment by Pyae-Phyo-Swe** [comment]
<img width="256" alt="image" src="https://user-images.githubusercontent.com/89895981/145331686-03c017ec-f2a6-42f7-9296-2a880fcb2554.png">
You can use the JS DOM under useEffect or whatever you want like this . 👉
`document.getElementById('permission-list').innerHTML=selectedPermission;`
Where selectedPermission is the hook.
